File: group_restrictions.js
$(function() {
	$("#group_restrictions").val(group_restrictions.map(function(entry) {
		return entry.grouping.join(", ")+": "+entry.restrictions.join(", ");
	}).join("\n"));
	$("#group_restrictions").change(function() {
		var lines = $("#group_restrictions").val().split(/\r\n|\r|\n/);
		group_restrictions = lines.map(function(line) {
			var parts = line.split(":");
			return {
				grouping: parts[0].split(",").map(s=>s.trim()).filter(s=>s),
				restrictions: parts[1].split(",").map(s=>s.trim()).filter(s=>s),
			};
		});
	});
});

File: group_restrictions.py
import re

def update_group_restrictions(group_restrictions):
    group_restrictions_str = "\n".join(
        [", ".join(entry['grouping']) + ": " + ", ".join(entry['restrictions']) for entry in group_restrictions]
    )
    
    # Simulating the change event
    lines = group_restrictions_str.splitlines()
    group_restrictions = [
        {
            'grouping': list(filter(None, [s.strip() for s in parts[0].split(",")])),
            'restrictions': list(filter(None, [s.strip() for s in parts[1].split(",")]))
        }
        for line in lines if line
        for parts in [line.split(":")]
    ]
    
    return group_restrictions